### PR TITLE
[2017.7] git.minimal installation fixes/additions for golden images

### PR DESCRIPTION
**Pull request by s0undt3ch**

### CODE REVIEW

File: _modules/pip.py [hunk 1]
```diff
@@ -29,29 +30,32 @@
...
 # Let's namespace all other functions from the pip module
 for name in dir(salt.modules.pip):
     attr = getattr(salt.modules.pip, name)
     if isinstance(attr, types.FunctionType):
-        if attr == 'install':
-            continue
-        if attr in globals():
+        if attr in ('install', 'freeze', 'list', 'list_', __get_pip_bin):
             continue
+        #if attr in globals():
```
Comment (Ch3LL) [inline _modules/pip.py]: can we remove the comments?

File: _modules/pip.py [hunk 2]
```diff
@@ -89,38 +94,53 @@ def get_pip_bin(bin_env, pip_bin_name=None):
...
+        log.debug('pip_bin_name resolved to: %s', pip_bin_name)
+
 
     if not bin_env:
         # not in / or c ignores the "root" directories as virtualenvs
         which_result = __salt__['cmd.which_bin']([pip_bin_name])
         if which_result is None:
-            raise CommandNotFoundError('Could not find a `pip` binary')
+            return _list_or_not(__get_pip_bin(bin_env))
+            #raise CommandNotFoundError('Could not find a `pip` binary')
```
Comment (Ch3LL) [inline _modules/pip.py]: another comment, can we remove?

File: _modules/pip.py [hunk 3]
```diff
@@ -89,38 +94,53 @@ def get_pip_bin(bin_env, pip_bin_name=None):
...
+        log.debug('bin_env(%s) is not a direactory', bin_env)
         if os.path.isfile(bin_env) or os.path.islink(bin_env):
+            log.debug('bin_env(%s) exists, returning it', bin_env)
+            cache_pip_version(bin_env)
             return _list_or_not(bin_env)
     else:
-        raise CommandNotFoundError('Could not find a `pip` binary')
+        log.debug('Could not find a pip binary with bin_env(%s) and pip_bin_name(%s)!!!!', bin_env, pip_bin_name)
+        return _list_or_not(__get_pip_bin(bin_env))
+        #if raise_error:
```
Comment (Ch3LL) [inline _modules/pip.py]: here too

**Comment by s0undt3ch** [comment]
Yup!

**Pull request merged**